src/schema/schema_stat.c

```diff
@@ -147,13 +147,27 @@ __wt_curstat_table_init(
 
     WT_ERR(__wt_scr_alloc(session, 0, &buf));
 
+    stats = &cst->u.dsrc_stats;
+
+    /*
+     * If gathering statistics for a simple table, retrieve the underlying file's statistics rather
+     * than going through the usual schema parsing flow. This avoids unnecessary overhead in the
+     * statistics gathering process.
+     *
+     * There will always be at least one column group for a table.
+     */
+    if (table->is_simple) {
+        WT_ERR(__wt_buf_fmt(session, buf, "statistics:%s", table->cgroups[0]->name));
+        ret = __wt_curstat_init(session, buf->data, NULL, cfg, cst);
+        goto done;
```

Comment:
If we have a `goto done` here now, I suggest the following to avoid any issue in the long run:
```suggestion
        WT_ERR(__wt_curstat_init(session, buf->data, NULL, cfg, CST));
        goto done;
```
What do you think @mm-ng @sulabhM?

Reply:
Yes, I'd be happy to put this change in.

Reply:
Sounds good, resolving the conversation to avoid confusion.